wl-sim/main/WL_Flash.cpp

```diff
@@ -0,0 +1,292 @@
+#include <cmath>
+#include "esp_log.h"
+#include "esp_err.h"
+
+#include "wl_sim.h"
+
+static const char *TAG = "WL_Flash";
+
+#define SECTOR_ERASE_ENDURANCE 100000
+
+size_t access_count = 0;
+size_t pos = 0;
+size_t move_count = 0;
+uint32_t cycle_count = 0;
+uint32_t erase_count[SECTOR_COUNT + 1] = {0};
+size_t last_erase_sector = 0;
+size_t restarted = 0;
+
+// key_len is half of bit length of address space
+// so 8bit key is for (up to) 16bit address space
+// 65K sectors is plenty: up to 32MB of 512B sectors
+// or 256MB of 4K sectors
+uint8_t keys[3] = {0};
+uint8_t B = 0, MSB = 0, LSB = 0;
+uint32_t feistel_calls = 0;
+uint32_t feistel_cycle_walks = 0;
+static bool feistel = false;
+
+void init_feistel()
+{
+    feistel = true;
+
+    uint16_t sector_count = SECTOR_COUNT;
+    ESP_LOGI(TAG, "%s: N=%u", __func__, sector_count);
+    ESP_LOGI(TAG, "sizeof(size_t)=%u", sizeof(size_t));
+
+    // B = log2(sector_count)
+    for (B = 0; sector_count; B++)
```

Comment:
FYI: to meet formatting standards of the IDF development rules, you can use $IDF_PATh/tools/format.sh _<c/cpp file>_

Reply:
got it